runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7881,8 +7881,83 @@ done:;
 	VMINLINE VM_BytecodeAction
 	withfield(REGISTER_ARGS_LIST)
 	{
+retry:
 		VM_BytecodeAction rc = EXECUTE_BYTECODE;
-		// TODO: Implement bytecode
+
+		U_16 index = *(U_16*)(_pc + 1);
+		J9ConstantPool *ramConstantPool = J9_CP_FROM_METHOD(_literals);
+		J9RAMFieldRef *ramFieldRef = ((J9RAMFieldRef*)ramConstantPool) + index;
+		UDATA const flags = ramFieldRef->flags;
+		UDATA const valueOffset = ramFieldRef->valueOffset;
+		j9object_t original = *(j9object_t *)(_sp + ((flags & J9FieldSizeDouble) ? 2 : 1));
+		j9object_t copy = NULL;
+		J9Class *objectClass = J9OBJECT_CLAZZ(_currentThread, original);
+
+		/* In a resolved field, flags will have the J9FieldFlagResolved bit set, thus
+		 * having a higher value than any valid valueOffset.
+		 *
+		 * This check avoids the need for a barrier, as it will only succeed if flags
+		 * and valueOffset have both been updated. It is crucial that we do not treat
+		 * a field ref as resolved if only one of the two values has been set (by
+		 * another thread that is in the middle of a resolve).
+		 */
+		if (J9_UNEXPECTED((flags <= valueOffset) || J9_ARE_NO_BITS_SET(flags, J9FieldFlagPutResolved))) {
+			/* Unresolved */
```

Comment:
Add more detail to this comment. What is unresolved, and why we need to resolve it?